app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java

```diff
@@ -1,19 +1,24 @@
 package com.example.parkingsystem.mvp.presenter;
 
-import com.example.parkingsystem.mvp.contract.ParkingContract;
+import com.example.parkingsystem.mvp.contracts.ParkingContract;
 
 public class ParkingPresenter implements ParkingContract.ParkingPresenter {
 
-    private ParkingContract.ParkingModel model;
-    private ParkingContract.ParkingView view;
+    private final ParkingContract.ParkingModel model;
+    private final ParkingContract.ParkingView view;
```

Comment:
this variables not have to be finals

Reply:
Done!

Reply:
If i'm not wrong the final is suggested by the AS. it is not wrong that appears as final I think

Reply:
Yeah it's suggested! But anyway I deleted on this and other files where they have the same problem. I only left the final in the dialog fragment's tag